### ISSUE TITLE
Document integrity signature for official client binaries on all platforms

### ISSUE THREAD

**Issue by japostoles** [owncloud/client#5081]
It would be helpful to have the signatures for the official binaries in the same way that source tarballs are signed.  Currently there's no way to verify that the desktop client binaries are official, untampered builds.

Any chance we can get the release binaries signed?  Thanks!

**Comment by japostoles** [comment]
I was referring to Mac in this case.  I didn't release that Mac .pkg files were signed.  I'm used to running gpg --verify on downloads that have signatures.  

Maybe this is redundant in this case?

After doing some research, I was able to verify using this command:

```
[MacBook:Downloads]$ pkgutil --check-signature ownCloud-2.2.2.3472.pkg
Package "ownCloud-2.2.2.3472.pkg":
   Status: signed by a certificate trusted by Mac OS X
   Certificate Chain:
    1. Developer ID Installer: ownCloud Inc. (9B5WD74GWJ)
       SHA1 fingerprint: 9E D0 FA D4 18 45 32 F1 92 B4 64 8E BE A2 33 8D 31 CA 13 17
       -----------------------------------------------------------------------------
    2. Developer ID Certification Authority
       SHA1 fingerprint: 3B 16 6C 3B 7D C4 B7 51 C9 FE 2A FA B9 13 56 41 E3 88 E1 86
       -----------------------------------------------------------------------------
    3. Apple Root CA
       SHA1 fingerprint: 61 1E 5B 66 2C 59 3A 08 FF 58 D1 4A E2 24 52 D1 98 DF 6C 60
```

Looks like you're right on this.  A sig file couldn't hurt, but it's definitely redundant.

Thanks!

**Comment by RealRancor** [comment]
Maybe something to document ( if its not already documented)?

**Comment by jnweiger** [comment]
Linux packages are signed by openSUSE build service with e.g. this key:
  http://download.opensuse.org/repositories/isv:ownCloud:/desktop/Ubuntu_16.04/Release.key

pub  1024D/BA684223 2012-02-08 isv:ownCloud OBS Project <isv:ownCloud@build.opensuse.org>
sig        BA684223 2015-02-09   [selfsig]
sig        6B9D6523 2012-02-08   [User-ID nicht gefunden]

It is "documented" at https://software.opensuse.org/download.html?project=isv:ownCloud:desktop&package=owncloud-client -- not sure, if that is also for download at owncloud.org somewhere.

**Comment by RealRancor** [comment]
Yeah, i though about the mac stuff to document. :-) I think thats not that obvious.

**Comment by jnweiger** [comment]
There is one sig on https://owncloud.org/install/#install-clients
One sig, but the context is three platforms: Windows, Mac, Linux. 
Not sure which one *that* is.

**Comment by RealRancor** [comment]
Thats the signature file for the sources:

https://download.owncloud.com/desktop/stable/owncloudclient-2.2.2.tar.xz
-> https://download.owncloud.com/desktop/stable/owncloudclient-2.2.2.tar.xz.asc

**Comment by japostoles** [comment]
Yeah I think the point here is that with any binary built from source that is distributed, clear instructions on verifying the package should be somewhere in the install guide.  Not only to verify that the package itself is signed, but also to ensure it's signed with the correct cert.  At a minimum the cert it should be signed with should be documented for comparison.

In this case, the signature file given for the sources isn't helpful.  What we need is something stating, on a Mac, the package should be signed by the trusted certificate `ownCloud Inc. (9B5WD74GWJ)`

This is different than the signing key used when signing sources (`ownCloud Client Team (Signing Key) E94E7B37`)

As described above, running `pkgutil --check-signature ownCloud-x.y.z.nnnn.pkg` will output the certificate chain that the package is signed with that should then be compared to whatever the official certificate is (looks like 9B5WD74GWJ).

**Comment by settermjd** [comment]
@japostoles to check that I'm correctly understanding what you're asking; do you only want instructions on verifying packages when building from source?
